enclave/core/sgx/calls.c

```diff
@@ -1444,3 +1453,38 @@ void oe_abort(void)
 
     oe_abort_with_td(td);
 }
+
+static uint64_t _page_fault_debug_handler(oe_exception_record_t* record)
+{
+    if (oe_is_enclave_debug_allowed() &&
+        record->code == OE_EXCEPTION_PAGE_FAULT)
+        OE_TRACE_WARNING("Page fault occured. The execution will be aborted if "
```

Comment:
This seems fine to me. The second pass handler that calls the registered-signal handler stitches the callstack to the point of exception.

Reply:
Can could print the backtrace here, but when the execution aborts, another backtrace will be printed.

Reply:
You could change the warning message to indicate where in the printed callstack the developer should look for SEGV.

Reply:
Do we know on the host-side whether the SEGV came from the enclave? If so, we could print a message describing how the enclave can be configured to print the backtrace for SEGV.

Reply:
Updated to print helper message.